internal/blobtesting/object_locking_map.go

```diff
@@ -60,7 +62,7 @@ func (s *objectLockingMap) getLatestForMutationLocked(id blob.ID) (*entry, error
 	}
 
 	if !e.retentionTime.IsZero() && e.retentionTime.After(s.timeNow()) {
-		return nil, errors.New("cannot alter object before retention period expires")
+		return nil, errors.WithStack(ErrBlobLocked)
```

Comment:
In the long term, we are going to move away from `pkg/errors`. It's deprecated.

Reply:
ah right, forgot about that thanks for the reminder. Well, at least until the switch this will give us stack trace info about where the error came from which is always sort of nice 😅